DiscordChatExporter.Core/Exporting/Writers/Html/MessageGroupTemplate.cshtml

```diff
@@ -81,7 +100,38 @@
 
             @{/* Right side */}
             <div class="chatlog__message-primary">
-                @if (isFirst)
+                @if (isSystemMessage)
+                {
+
+                    // system messages are grouped even if the message author is different
+                    // that's why we have to update the user values with the author of the current message
+                    userMember = Model.ExportContext.TryGetMember(message.Author.Id);
+
+                    userColor = Model.ExportContext.TryGetUserColor(message.Author.Id);
+
+                    userNick = message.Author.IsBot
+                        ? message.Author.Name
+                        : userMember?.Nick ?? message.Author.Name;                    
```

Comment:
Can we just move the whole resolving logic from the top part to here, instead of overwriting the variables?

Reply:
Hmm that thought crossed my mind too, but I thought to keep it the complexity low and readability high it makes more sense to have it at the top. (there is also a small performance gain I am pretty sure since most messages are not system messages)

The part in the loop is getting quite complex already so I only moved in the necessary parts.